plugins/modules/oneandone_server.py

```diff
@@ -553,9 +554,15 @@ def _auto_increment_hostname(count, hostname):
     name-02, name-03, and so forth.
     """
     if "%" not in hostname:
-        hostname = "%s-%%01d" % hostname  # noqa
+        hostname = f"{hostname}-%01d"
 
-    return [hostname % i for i in range(1, count + 1)]
+    hostname_re = re.compile(r"^(.*)%(0\d+)d$")
+
+    if not hostname_re.match(hostname):
+        raise ValueError("Hostname numbering format must be pattern `%0<n>d` with <n> being the number digits wanted")
+
+    hostname_template = hostname_re.sub(r"\1{:\2}", hostname)
+    return [hostname_template.format(i) for i in range(1, count + 1)]
```

Comment:
While the documentation only mentions `%02d` and `%03d`, any `%` formatting specifier can be used that can format an integer, like `%s`. I don't think it's a good idea to re-implement `%` here, so why not simply keep the original `return [hostname % i for i in range(1, count + 1)]`? Everything else is likely a breaking change.